File: String/Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        temp = dict()
        for i, num in enumerate(nums):
            if num in temp and abs(i-temp[num]) <= k:
                return True
            temp[num] = i

File: String/Contains Duplicate II.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        map<int, int> temp;
        for(int i=0; i<nums.size(); i++){
            if((temp.find(nums[i]) != temp.end()) && (i-temp[nums[i]]) <= k){
                return true;
            }
            temp[nums[i]] = i;
        }
        return false;
    }
};